opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/frontend/AccountEditActivity.java

```diff
@@ -189,6 +201,36 @@ private void delete() {
     }
 
     @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent idata) {
+        super.onActivityResult(requestCode, resultCode, idata);
+
+        //TODO: Rewrite this for the new SearchField implementation
+        // Barcode
+        BarcodeScanIntegrator.ScanResult scanResult = BarcodeScanIntegrator
+                .parseActivityResult(requestCode, resultCode, idata);
+        if (resultCode != RESULT_CANCELED && scanResult != null) {
+            if (scanResult.getContents() == null) {
+                return;
+            }
+            else if (scanResult.getContents().length() < 3) {
+                return;
+            }else {
+
+                account.setName(scanResult.getContents());
+                etName.setText(account.getName());
+            }
+
+
+        }
+    }
+
+    @Override
+    public void scanBarcode() {
```

Comment:
You don't really need this method as it won't be called from outside, would you mind just moving this code up into your onClick handler?